git/git_intro.qmd

```diff
@@ -60,6 +60,26 @@ Staging in Git involves adding new, modified, or deleted files to a staging area
   The resulting commit will replace the incomplete one, and it will look like we committed the changes to hello.py and main.py in a single snapshot.
 
 ## Git push/pull
+You can use `git push` to sync a remote repository with the changes you've done locally. 
+The most basic example would be that you've first cloned a repository with `git clone` then made some changes in that local copy and want to update the original remote repository.  
+Similarly, if for example, someone else made changes to the remote and you want to incorporate those changes into your local copy you will run `git pull` to make sure you are up to date with the changes in the remote repository before working on your local copy. 
 
 ## Git merge and git rebase
+Git merge and git rebase can be said to be used to solve similar things.  
+When working on a feature in a separate branch while someone else updates the main branch you often want to incorporate the new changes from the main branch into your feature branch.  
+This could be done with merge like this: 
```

Comment:
I think before `git merge main` it would be necessary to pull the changes that someone made on the main branch of the remote repository on your local repository, and then `git merge main`

Reply:
Thanks, tried to clarify now in the text.